### ISSUE TITLE
Style variations not merging correctly

### ISSUE THREAD

**Issue by richtabor** [Automattic/wp-calypso#96036]
Outside of WordPress.com, you can take [this pattern markup](https://gist.github.com/richtabor/e25d9e316e51f39495425c86b425a49e), activate the latest version of [Assembler](https://github.com/Automattic/themes/tree/trunk/assembler) and apply the Noir color palette—all colors are properly merged and accounted for. When applying a style variation in the editor, everything seems to works fine. However, when previewing that page (or refreshing the editor after saving), the computed styles are no longer correct. 

It’s likely more than just the button block is affected, but I’ve narrowed down what’s happening output-wise. 

### The problem

The parent theme.json file establishes a [styles.elements.button.hover](https://github.com/Automattic/themes/blob/trunk/assembler/theme.json#L876-L881) value like this: 

```
"button": {
        ":hover": {
              "color": {
                  "background": "color-mix(in srgb, var(--wp--preset--color--theme-5) 85%, #000)"
               }
         }
}
```

The Noir variation—and many others—establish their own button hover value, [like this](https://github.com/Automattic/themes/blob/trunk/assembler/styles/colors/09-noir.json#L108-L121).

```
 "elements": {
            "button": {
                ":hover": {
                    "color": {
                        "background": "color-mix(in srgb, var(--wp--preset--color--theme-4) 85%, #FFF)",
                        "text": "var(--wp--preset--color--theme-3)"
                    }
                }
            }
        },
...
```

The result after saving the editor (only on WordPress.com) is the output below: 

```
:root :where(.wp-element-button:hover, .wp-block-button__link:hover) {
	background-color: color-mix(in srgb, var(--wp--preset--color--theme-5) 85%, #000);
	color: var(--wp--preset--color--theme-3);
}
```

When it should use the Noir variation's styles exclusively: 

```
:root :where(.wp-element-button:hover, .wp-block-button__link:hover) {
    background-color: color-mix(in srgb, var(--wp--preset--color--theme-4) 85%, #FFF);
    color: var(--wp--preset--color--theme-3);
}
```

#### Current 

https://github.com/user-attachments/assets/01b01322-e6c7-49c9-9ab8-aacb47a0e919

#### Expected

https://github.com/user-attachments/assets/089864ae-8366-4067-a1ff-3c9ac6145865

Note: If proxied, you can see [this test site](https://richptabor64.wordpress.com/) I created with the issue persisting on the main button in the hero area.

**Comment by BogdanUngureanu** [comment]
I was able to reproduce it on a Dotcom Simple Site (on Atomic it works fine). I also made the tests with the same Gutenberg version and it's not from there. 

I did some debugging and the only thing I observed was that on save, we make a request to `wp/v2/sites/site-id/global-styles/post-id`. On a self hosted site, the payload we pass contains `settings` and `styles`
On a dotcom, we seem to only pass the `styles`  - not sure if this is what's causing the problem. The payload is also different than a self hosted one.
I don't think it's a front-end problem (like we had in the past). I think in this case, after the variation is saved, the response (styles) is loaded in the editor again and breaks the hover:
- Click on the Noir variation
- Notice that the hover is working as expected
- Click save
- The hover is broken
- Select noir again
- Notice is highlighted in the panel
- Click Save immediately
- Notice that Noir variation is no longer selected in the GS panel

**Comment by richtabor** [comment]
Do you think it's related to the formatting of this value? `color-mix(in srgb, var(--wp--preset--color--theme-4) 85%, #FFF)`

**Comment by BogdanUngureanu** [comment]
Opened D165630-code that fixes the issue.

**Comment by dsas** [comment]
@BogdanUngureanu also opened https://github.com/WordPress/gutenberg/issues/66799 to fix this issue upstream.

**Comment by Robertght** [comment]
📌 ACTIONS
- Triaged
- Marking as Core fix needed.

**Comment by BogdanUngureanu** [comment]
Deployed a fix with D165630-code
